src/components/dialogs/filter/expert/expert-filter-utils.ts

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import {
+    defaultOperators,
+    QueryValidator,
+    RuleGroupType,
+    RuleGroupTypeAny,
+    ValidationMap,
+} from 'react-querybuilder';
+import { RuleType } from 'react-querybuilder/dist/cjs/react-querybuilder.cjs.development';
+import { DataType, fields, operatorType } from './expert-filter-constants';
+import { IntlShape } from 'react-intl';
+import { EMPTY_GROUP, INCORRECT_RULE } from 'components/utils/field-constants';
+import { EMPTY_RULE } from '../../../utils/field-constants';
+
+const getDataType = (fieldName: string) => {
+    const generatorField = fields().GENERATOR.find(
+        (field) => field.name === fieldName
+    );
+    const loadField = fields().LOAD.find((field) => field.name === fieldName);
+
+    const field = generatorField || loadField;
+    return field?.dataType;
+};
```

Comment:
in my opinion, this is a utility method which should getDataType by filedName and equipmentType. By this way, we can avoid maintain this method when supporting another equipment type in the future.. EquipmentType should be passed from the concrete usage context